dvc/remote/local/__init__.py

```diff
@@ -195,7 +195,7 @@ def already_cached(self, path_info):
         if not current_md5:
             return False
 
-        return not self.changed_cache(current_md5)
+        return not self.cache.changed_cache(current_md5)
```

Comment:
again, what's the difference here. All this relationship between remote, cache, and output is very confusing tbh.

Reply:
@shcheklein , it is confusing, indeed. This solve the same problem explained above.

Reply:
@mroutis So why not `self.changed_cache`? It should do the same thing, no? Why do we need to access self.cache.changed_cache directly?

Reply:
`self` is a remote with no `path_info`, due to the way outputs are instantiated:
```python
        self.remote = remote or self.REMOTE(self.repo, {})
```

If we call `changed_cache` directly it will fail on this one:
```python
    def checksum_to_path_info(self, checksum):
        return self.path_info / checksum[0:2] / checksum[2:]
```

Reply:
@mroutis I guess you meant that
```
        return not self.cache.changed_cache(current_md5)
```
is different from
```
    def changed_cache(self):                           
        if not self.use_cache or not self.checksum:    
            return True                                
                                                       
        return self.cache.changed_cache(self.checksum) 
```
because in the self.changed_cache we are using self.checksum wich might not match current_md5?

Reply:
@efiop , where did you get the `self.use_cache` and `self.checksum`? I'm not sure if we are talking about the same `changed_cache` :sweat_smile: there's one in `output/base.py` and another one in `remote/base.py`. I'm talking about latter one. (this change is on `remote/base.py` -- just clearing it out of the way, it might result confusing)

Reply:
@mroutis Aah, indeed, i was looking at `output` for some reason. Sorry! So looks like we've had a bug here before, right?